spvc/test/run_spirv_cross_tests.py

```diff
@@ -25,152 +25,189 @@
 import tempfile
 
 
-def log_unexpected_successes(script_args, successes):
-    if not len(successes):
-        log_string = 'Encountered 0 unexpected successes'
-    else:
-        log_string = 'Encountered {} unexpected success(es):\n'.format(
-            len(successes))
-        successes = ['\t{}'.format(success) for success in successes]
-        log_string += '\n'.join(successes)
-
-    if script_args.log:
-        script_args.log.write(log_string + '\n')
-        script_args.log.flush()
-    else:
+class TestEnv:
+    """Container for cross-test environmental data and operations."""
+
+    def __init__(self, script_args):
+        """Takes in the output of ArgumentParser.parse_args()"""
+        # TODO(bug 737): Add new log data structure.
+        self.dry_run = script_args.dry_run
+        self.give_up = script_args.give_up
+        self.cross_dir = script_args.cross_dir
+        self.spvc = os.path.join(script_args.build_dir, 'spvc', 'spvc')
+        self.glslang = os.path.join(
+            script_args.build_dir, 'third_party', 'glslang', 'StandAlone', 'glslangValidator')
```

Comment:
This is a strong dependence on the structure of the build tree.
We have to support wildly configurable build trees, e.g. with SPIRV-Tools,  glslang being built out of tree.

Also, having one arg per executable we need makes dependencies explicit. It's clear  to the build system what files are needed before we can run the script.

I would really prefer to keep separate args: one per required executable.

Now, for usability, I suggest making a shell script that wraps this script, since you yourself know your build tree configration.

Reply:
Done